### ISSUE TITLE
Function not implemented

### ISSUE THREAD

**Issue by mictadlo** [maxplanck-ie/snakepipes#296]
Hi, 
I used `HiC dbg2olc40x -i /scratch/waterhouse_team/benth/dbg2olc-40x/snakepipes/snakePipes/workflows/HiC/hic-fq  -o /scratch/waterhouse_team/benth/dbg2olc-40x/snakepipes/snakePipes -c defaults.yaml`  but I got this error:

```
Failed to create temp dir under temp path prefix (/data/extended/)! Try fallback: /scratch/waterhouse_team/benth/dbg2olc-40x/snakepipes/snakePipes/ ...
Building DAG of jobs...
Using shell: /bin/bash
Provided cluster nodes: 5
Job counts:
        count   jobs
        4       FASTQ
        1       all
        2       build_matrix
        2       call_tads
        2       compute_thresholds
        2       correct_matrix
        2       diagnostic_plot
        4       map_fastq_single_end
        1       multiQC
        20

[Wed Oct 31 11:18:00 2018]
localrule FASTQ:
    input: /scratch/waterhouse_team/benth/dbg2olc-40x/snakepipes/snakePipes/workflows/HiC/hic-fq/N_Ben_HiC4.R1.fastq.gz
    output: FASTQ/N_Ben_HiC4.R1.fastq.gz
    jobid: 5
    wildcards: sample=N_Ben_HiC4.R, read=1


[Wed Oct 31 11:18:00 2018]
localrule FASTQ:
    input: /scratch/waterhouse_team/benth/dbg2olc-40x/snakepipes/snakePipes/workflows/HiC/hic-fq/N_Ben_HiC4.R2.fastq.gz
    output: FASTQ/N_Ben_HiC4.R2.fastq.gz
    jobid: 11
    wildcards: sample=N_Ben_HiC4.R, read=2


[Wed Oct 31 11:18:00 2018]
localrule FASTQ:
    input: /scratch/waterhouse_team/benth/dbg2olc-40x/snakepipes/snakePipes/workflows/HiC/hic-fq/N_Ben_HiC2.R1.fastq.gz
    output: FASTQ/N_Ben_HiC2.R1.fastq.gz
    jobid: 9
    wildcards: sample=N_Ben_HiC2.R, read=1


[Wed Oct 31 11:18:00 2018]
localrule FASTQ:
    input: /scratch/waterhouse_team/benth/dbg2olc-40x/snakepipes/snakePipes/workflows/HiC/hic-fq/N_Ben_HiC2.R2.fastq.gz
    output: FASTQ/N_Ben_HiC2.R2.fastq.gz
    jobid: 17
    wildcards: sample=N_Ben_HiC2.R, read=2

touch: touch: touch: touch: setting times of 'FASTQ/N_Ben_HiC4.R1.fastq.gz'setting times of 'FASTQ/N_Ben_HiC2.R1.fastq.gz'setting times of 'FASTQ/N_Ben_HiC2.R2.fastq.gz'setting times of 'FASTQ/N_Ben_HiC
4.R2.fastq.gz': Function not implemented: Function not implemented: Function not implemented: Function not implemented
```
What did I miss?

Thank you  in advance,

Michal

**Comment by katsikora** [comment]
Hi Michal, what is your OS? Are you working on a Mac?

Also, the first line of your stdout tells me you didn't update the "tempdir" key in the defaults.yaml dictionary and you're using *our* local setting. Snakemake than uses your system default TMP (/scratch/waterhouse_team/benth/dbg2olc-40x/snakepipes/snakePipes/...) - make sure this has enough space for what you're trying to do, or update that yaml.
You should have be able to find it under: snakePipes/shared/defaults.yaml .

This setting is not related to what causes your error.

Best,

Katarzyna

**Comment by dpryan79** [comment]
You seem to be using an odd file system such that the Linux `touch` command doesn't work on it. You'll need to fix that.

**Comment by katsikora** [comment]
Hi Michal,

so as Devon says there is some issue with how the touch command is working on your system.

Perhaps you can try the following:

Check how rule FASTQ is implemented - you should be able to find it in snakePipes/shared/rules/FASTQ.snakefile. 
Consider:
a) deleting  the string "&& touch -h {output}" in the last line of that snakefile
b) deleting just the "-h" flag 
This flag doesn't seem to be implemented on SUSE v.11.3, I don't know how about your version --->> try "man touch" on your terminal. On CentOS, this flag ensures the timestamp is changed on the symlink target only, not affecting the link source file. 

You will have to reinstall snakePipes from your local (modified) copy then.

If none of this works for you, you might want to speak to your system admin.

HTH,
Best,

Katarzyna

**Comment by steffenheyne** [comment]
it seems the touch option "-r --reference=FILE" flag is implemented on SUSE as well, so we might switch to this more generic one and use as file e.g. the config.yaml inside the output directory

**Comment by mictadlo** [comment]
When would I be able to test it?

**Comment by vivekbhr** [comment]
@mictadlo It's fixed now in the develop branch..

**Comment by mictadlo** [comment]
I did `snakepipes> git checkout develop` but still got the same problem. Do I have to do anything else to get the fix?

Thank you in advance,

Michal

**Comment by dpryan79** [comment]
You’ll need to install it.

**Comment by mictadlo** [comment]
how?

**Comment by mictadlo** [comment]
Hi all,
I have done the following steps but ran into this problem:

```
> pip install --user --upgrade git+https://github.com/maxplanck-ie/snakepipes@develop
> createIndices --local -j 8 --genomeURL /scratch/waterhouse_team/benth/dbg2olc-40x/salsa/scaffolds_FINAL.fasta -o snakePipes-output --tempdir snakePipes-tmp dbg2olc40x
Traceback (most recent call last):
  File "/work/waterhouse_team/miniconda2/envs/snakePipes/bin/createIndices", line 104, in <module>
    main()
  File "/work/waterhouse_team/miniconda2/envs/snakePipes/bin/createIndices", line 94, in main
    snakemake_cmd = cf.commonYAMLandLogs(baseDir, workflowDir, defaults, args, __file__)
  File "/home/lorencm/.local/lib/python3.6/site-packages/snakePipes/common_functions.py", line 374, in commonYAMLandLogs
    shutil.copyfile(orgyaml, organismYAMLname)
  File "/work/waterhouse_team/miniconda2/envs/snakePipes/lib/python3.6/shutil.py", line 120, in copyfile
    with open(src, 'rb') as fsrc:
FileNotFoundError: [Errno 2] No such file or directory: 'dbg2olc40x'
```
Did the syntax change?

Thank you in advance,

Michal

**Comment by dpryan79** [comment]
That particular bug is fixed in a pull request I expect to merge today. I need to test the last 3 pipelines first.

**Comment by mictadlo** [comment]
Hi, 
when do you think I can use it?

Thank you in advance,

Michal

**Comment by steffenheyne** [comment]
the fixes for the original topic as well as the createIndices bug are in develop branch now. you can try that as you did before. A new release is going to happen probably this week. We want to merge in first some changes about sample sheet handling

**Comment by steffenheyne** [comment]
we released  version 1.1.1 that should fix all of the issues
